subprojects/platform-native/src/main/java/org/gradle/language/nativeplatform/internal/IncludeDirectives.java

```diff
@@ -27,8 +29,8 @@
     List<Include> getAll();
     List<Include> getIncludesOnly();
 
-    List<Macro> getMacros();
-    List<MacroFunction> getMacrosFunctions();
+    Multimap<String, Macro> getMacros();
```

Comment:
I'm not sure if this change is a good idea. I was debating between using the `Multimap` directly or creating `MacroCollection` and `MacroFunctionCollection` class instead to encapsulate the implementation detail. Or a single collection class that takes a `Named` type generic which would require `Macro` and `MacroFunction` to extends from `Named`.

Reply:
Personally, I'd have something like `Iterable<Macro> getMacros(String name)` so that the implementation of `IncludeDirectives` can do whatever indexing makes sense for the directives it contains (eg we might have  different implementations for 'none', 'one' and 'many', and perhaps a composite implementation too).

Reply:
Good point, I will change that.